### PR TITLE
Add extension to generate data tables from SQL

### PR DESCRIPTION
**Pull request by simonwo**
We now download the tariff open data file and execute any embedded SQL blocks in the content against it, returning them as tables in the rendered documentation.

This means instead of specifying a static data table:

```markdown
|     Code    |     Description    |   
|-|-| 
|     9    |     National   document    |      
|     A    |     Certificate of authenticity    |     
|     C    |     Other   certificates    |    
|     D    |     Anti-dumping, countervailing   document    |   
|     E    |     Export   certificate, licence or document from country of origin    |    
|     H    |     HANDI,   LOOMS certificate    |     
|     I    |     Surveillance   certificate, licence or document issued by one of the Member States    |    
|     K    |     Tariff   quota    |    
|     L    |     Import   certificate, licence or document    |    
|     N    |     UN/EDIFACT   certificates    |     
|     P    |     Ingredients    |   
|     R    |     Export   refunds    |     
|     U    |     Proofs   of origin    |    
|     X    |     Export   licence    |     
|     Y    |     Particular   provisions    |     
|     Z    |     More   certificates    | 
```

we can instead just write the SQL statement used to generate it:

```sql
SELECT certificate_types.sid AS "Code",
       certificate_types.description AS "Description",
       COUNT(version_group_id) AS "Usage count"
FROM certificates
    INNER JOIN common_tracked_models ON certificates.trackedmodel_ptr_id = common_tracked_models.id
    INNER JOIN certificate_types ON certificates.certificate_type_id = certificate_types.trackedmodel_ptr_id
GROUP BY certificate_types.sid, description
ORDER BY certificate_types.sid
```

When the documentation is rendered, the SQL will be executed and the results embedded in the page as a table. This means the data tables stay current as the data changes, and include tricky-to-update stats like usage counts and last-used dates.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -84,6 +87,12 @@ OUTPUT_IMAGES.zip(IMAGE_FILES).each do |output, input|
   end
 end
 
+file LOCAL_DATABASE do
+  require 'open-uri'
+  rake_output_message "curl '#{SOURCE_DATABASE}' > #{LOCAL_DATABASE}"
```
Comment (paulpepper-trade) [inline Rakefile:92]: Thinking about things that could go wrong, like networking and filesystem access... Would `curl` and redirect errors be reasonably clear to readers of generated logs, or would it be useful to enhance that in some way?

**Review by paulpepper-trade — APPROVED** [review]
Very nice!

File: Rakefile (hunk 1 above)
Comment (simonwo) [reply Rakefile:92]: Hm, this should just print any network error encountered from within the Ruby stack. I don't think there's much more I can do to improve that. It honours redirects though.

File: lib/tariffsql.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+require 'sqlite3'
+
+DB = SQLite3::Database.new(ENV['LOCAL_DATABASE'] || 'tariff.sqlite')
+
+STATEMENTS = Hash.new do |hash, sql|
+  hash[sql] = DB.prepare sql
+end
+
+RESULTS = Hash.new do |hash, sql|
+  STDERR.puts sql
```
[the remark this answers is not in the document]
Comment (simonwo) [reply lib/tariffsql.rb:10]: Yes!

**Pull request merged**